src/server/pps/server/pipeline_controller.go

```diff
@@ -351,16 +352,15 @@ func (op *pipelineOp) rcIsFresh() bool {
 	rcName := op.rc.ObjectMeta.Name
 	rcPachVersion := op.rc.ObjectMeta.Annotations[pachVersionAnnotation]
 	rcAuthTokenHash := op.rc.ObjectMeta.Annotations[hashedAuthTokenAnnotation]
-	rcSpecCommit := op.rc.ObjectMeta.Annotations[specCommitAnnotation]
+	rcPipelineVersion := op.rc.ObjectMeta.Annotations[pipelineVersionAnnotation]
```

Comment:
Now we decide what to do in the pipeline controller based off the pipeline version rather than the spec commit ID.